### PR TITLE
Add repeals made

### PR DESCRIPTION
**Pull request by goose-life**
ref #1930

### CODE REVIEW

**Comment by actlikewill** [comment]
@goose-life I've have made updates to this PR, here is a demo https://www.loom.com/share/b324e6f676a145ffa6062a342817e183?sid=0b08ef62-3184-4424-8129-d9420983dbdd

File: indigo_app/views/works.py [hunk 1]
```diff
@@ -165,7 +184,27 @@ def form_valid(self, form):
                     task.locality = self.work.locality
                     task.save()
 
-            # signals
+            work_updated = True
+
+        # update repeals made
```
Comment (actlikewill) [inline indigo_app/views/works.py:189]: move the validation logic the forms  form_valid method and the forms save method, so validation runs once on the form

Comment (actlikewill) [reply indigo_app/views/works.py:189]: also overide form has_changed method to check if formset has changed

**Comment by actlikewill** [comment]
Demo with remove button: https://www.loom.com/share/159bdfc630f6487581fee9b55a7b368a?sid=1c314d8b-2e74-4459-ba84-2e4ca89e849d

**Comment by goose-life** [comment]
@actlikewill in that last demo, there's no repeal date? is there one under the hood that's just not displaying? 
(It shouldn't be possible to add a repeal without a repeal date, to be clear)

File: indigo_app/forms.py [hunk 2]
```diff
@@ -897,3 +933,30 @@ class FindPubDocForm(forms.Form):
...
+    def save(self):
+        work = self.cleaned_data['repealed_work']
+        if self.cleaned_data.get('DELETE'):
+            work.repealed_by = None
+            work.repealed_date = None
+            work.save()
+        else:
+            work.repealed_by = self.work
+            work.repealed_date = self.cleaned_data['repealed_date']
+            work.save()
```
Comment (actlikewill) [inline indigo_app/forms.py:962]: @longhotsummer I've moved most of the logic into the form. Feels much neater

File: indigo_app/views/works.py [hunk 3]
```diff
@@ -1318,3 +1318,44 @@ def get_context_data(self, **kwargs):
...
+class WorkFormRepealsMadeView(WorkViewBase, TemplateView):
+    form_class = RepealMadeForm
+    template_name = 'indigo_api/_work_form_repeals_made_form.html'
+
+    def post(self, request, *args, **kwargs):
+        return self.get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        RepealMadeBaseFormSet = formset_factory(RepealMadeForm, extra=0, can_delete=True)
```
Comment (longhotsummer) [inline indigo_app/views/works.py:1330]: Minor: you set this up in three places -- the best place to do it is in forms.py, just beneath RepealMadeForm,.

**Review by longhotsummer — APPROVED** [review]
Looks great

**Review by longhotsummer — APPROVED** [review]

File: indigo_app/forms.py [hunk 4]
```diff
@@ -897,3 +932,30 @@ class FindPubDocForm(forms.Form):
...
+
+    def is_repealed_work_saved(self):
+        return self.repealed_work_obj.repealed_by == self.work
+
+    def save(self):
+        work = self.cleaned_data['repealed_work']
+        if self.cleaned_data.get('DELETE'):
+            work.repealed_by = None
+            work.repealed_date = None
+            work.save()
```
Comment (goose-life) [inline indigo_app/forms.py:957]: I feel like this (and the save below) should be save_with_revision?

Comment (longhotsummer) [reply indigo_app/forms.py:957]: agreed - @actlikewill pls add in your next pr

**Pull request merged**